sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2383,6 +2383,17 @@ object functions {
     RegExpExtract(e.expr, lit(exp).expr, lit(groupIdx).expr)
   }
 
+  /**
+   * Extract all specific group matched by a Java regex, from the specified string column.
+   * If the regex did not match, or the specified group did not match, an empty array is returned.
```

Comment:
The behavior seems to be
1. If the regex does not match, return an empty array
2. if the specified group does not match, put an empty string to the result array.

Can we document the behavior in SQL expression? And can you verify this is the standard behavior in other databases?

Reply:
2. should throw a IllegalArgumentException. 
https://github.com/apache/spark/pull/27508
the behavior of Hive is :
`FAILED: SemanticException [Error 10014]: Line 1:7 Wrong arguments ‘2’: org.apache.hadoop.hive.ql.metadata.HiveException: Unable to execute method public java.lang.String org.apache.hadoop.hive.ql.udf.UDFRegExpExtract.evaluate(java.lang.String,java.lang.String,java.lang.Integer) on object org.apache.hadoop.hive.ql.udf.UDFRegExpExtract@2cf5e0f0 of class org.apache.hadoop.hive.ql.udf.UDFRegExpExtract with arguments {x=a3&x=18abc&x=2&y=3&x=4:java.lang.String, x=([0-9]+)[a-z]:java.lang.String, 2:java.lang.Integer} of size 3`

Reply:
let's document the behavior clearly.

Reply:
OK